### ISSUE TITLE
E022-error after deleting bookmark in Nextcloud -> no sync

### ISSUE THREAD

**Issue by BeatVonwil** [marcelklehr/floccus#522]
**Describe the bug**
when I delete a bookmark in Nextcloud, firefox add-on floccus synchronisation is broken with error E022.

**To Reproduce**
1. Situation:
Nextcloud (with Bookmarks) is set up
Floccus on Firefox is installed
account is created and initial sync complete

2. Manipulation:
I deleted a Bookmark in Nextcloud (in bookmark root folder)

3. Result:
Sync with floccus always stops with E022
>>E022: Ordner #-1[undefined] enhält angeblich ein nicht existierendes Lesezeichen: 57<<**Expected behavior**

4. Single solution:
Delete account in floccus and create it again.
Reset with >>Neue Synchronisierung von Grund auf auslösen<< does not solve the issue.

The intire behaviour is reproduceable.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Win10
 - Browser Firefox 76.0
 - Browser Version: 76.0
 - Floccus version: 4.1.0
 - Floccus sync method: Nextcloud Bookmarks (>>Nextcloud Bookmarks - 
Die Option "Nextcloud Bookmarks" ist kompatibel mit mindestens Version v0.14 der Bookmarks app.<<)

**Server (please complete the following information):**
 - Nextcloud version: 18.0.4
 - Bookmarks app version: 3.0.10

Note: Please write down the actual version numbers instead of writing 'latest'.

**Debug log**
2020-05-06T15:06:44.276Z Fetching bookmarks
2020-05-06T15:06:45.335Z Received bookmarks from server [   - #1[https://xxxxxxxxxxxxxxxxxxxxxxxxxxx.xxx](https://xxxxxxxxxxxxxxxxxxxxxxxxxxx.xxx) parentId: 2 ]
2020-05-06T15:06:46.391Z Received initial folders from server (may be incomplete) [ { id: 2,
    title: 'Folder1',
    userId: 'User1',
    parent_folder: 1,
    children: [] } ]
2020-05-06T15:06:47.505Z Received initial children order from server (may be incomplete) [ { type: 'folder', id: 2, children: [ [Object] ] },
  { type: 'bookmark', id: 57 } ]
2020-05-06T15:06:47.507Z [Error: E022: Ordner #-1[undefined] enhält angeblich ein nicht existierendes Lesezeichen: 57]
2020-05-06T15:06:47.507Z Syncing failed with E022: Ordner #-1[undefined] enhält angeblich ein nicht existierendes Lesezeichen: 57

- [ X] Debug log provided

**Additional context**

**Comment by BeatVonwil** [comment]
Solved!
I manually deleted bookmark '57' from Nextclouds mySQL DB. Then synchronization ran entirely.
Deleting a bookmark in Nextcloud maybe left an inconsistent record.

**Comment by marcelklehr** [comment]
Did you delete that bookmark with v3.0.10 or before updating?

**Comment by marcelklehr** [comment]
I cannot replicate this with v3.0.10.

**Comment by BeatVonwil** [comment]
I started from scratch with 3.0.10. I never had an earlier version.
I can reproduce it. It seems to be an error of app (nexcloud bookmark app 3.0.10). The _delete_ function seems to be incomplete.

What I did.
1. I have an NC account with 2 folders and 3 bookmarks
2. synced with floccus. ok.
3. in Firefox bookmark manager I copied other folders with ~50 bookmarks into NC folder
4. synced with floccus. ok.
5. In app I selected all bookmarks without tags and deleted them. I did not delete the newly created folders.
6. synced with floccus. Error E022.

7. Solution: Also deleting newly created, empty folders in app cleaned the issue. Syncing with floccus runs successfully.

I parallely verified all in DB. After deletion, _cc_bookmarks_tree_ lists bookmarks that do no longer exist. Deleting the directory does not clean _cc_bookmarks_tree_. But since the directory does no longer exist, the corpses will not be queried.
